protocols/kad/src/query/peers/closest/disjoint.rs

```diff
@@ -452,16 +452,16 @@ mod tests {
     use std::iter;
 
     impl Arbitrary for ResultIter<std::vec::IntoIter<Key<PeerId>>> {
-        fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        fn arbitrary(g: &mut Gen) -> Self {
             let target = Target::arbitrary(g).0;
-            let num_closest_iters = g.gen_range(0, 20 + 1);
-            let peers = random_peers(g.gen_range(0, 20 * num_closest_iters + 1), g);
+            let num_closest_iters = g.gen_range(0..20 + 1);
+            let peers = random_peers(g.gen_range(0..20 * num_closest_iters + 1));
 
             let iters: Vec<_> = (0..num_closest_iters)
                 .map(|_| {
-                    let num_peers = g.gen_range(0, 20 + 1);
+                    let num_peers = g.gen_range(0..20 + 1);
                     let mut peers = peers
-                        .choose_multiple(g, num_peers)
+                        .choose_multiple(&mut rand::thread_rng(), num_peers)
```

Comment:
Would `Gen::choose` work here? That way the `Arbitrary` implementation is still deterministic.

Reply:
We need a non-repeating random list.